### ISSUE TITLE
Trimming video raise sometimes AudioTrim: last sample beyond clip end error

### ISSUE THREAD

**Issue by sosie-js** [vapoursynth/vapoursynth#1084]
Hello all,

I try to reproduce the problem that occurs when trimming videos that has audio.
I used [a script to make the comparison between differents vs plugins videonode props and methods such as ffmpeg and ffprobe or vspipe info](https://github.com/sosie-js/python-bestsource/blob/1.0.0/tests/VTS_01_1-trim100.py).

![image](https://github.com/user-attachments/assets/3c8ef3f5-24d0-44d0-8a0d-8041d238d879)

What you will notice is that frame duration varies between tools. Most of them agree on num_frames to 535
with the provided video. 

I have had a surprise having "AudioTrim: last sample beyond clip end" error after trimming mostly on the last part trimmed
Then came me the idea to use trim on the length on the video thinking it should has no effect to find where this
overflow comes from. 

here is my attempt that suggest me frame_num is shorter by one deduced from the case 1. The case 2 suggest to reduce
video by 2 frames to fix the problem. What is correct to have a lossy trim in such situation? Thanks for your lights.

```python
clip=BSSource(input)
#type(clip) is Clip , class code was adapted from Clip Class of _AI_ https://forum.doom9.org/showthread.php?t=184300
video=Video(clip)

print("\n--- Case1: fake trim from 0 to  video.num_frames-1="+str(video.num_frames-1))
a=video.num_frames
adjust=0

#Fix fails vapoursynth.Error: AudioTrim: last sample beyond clip end
while True:
    try:
        clip=Trim(0,  video.num_frames-1-adjust) 
        audio_cut =get_audio_last_trimmable(clip)
        break;
    except:
        adjust=adjust+1
```

with get_audio_last_trimmable(clip) 

```python
def to_samples(clip, frame):
    return int((clip.audio.sample_rate/clip.video.fps)*frame)    

def get_audio_last_trimmable(clip):
    adjust=0
    video=clip.video
    while True:
        try:
            start=0
            last=video.num_frames-1
            afirst  = to_samples(clip, first)    if first  is not None else None
            alast   = to_samples(clip, last)-1-adjust if last  is not None else None
            alength= None
            audio.std.AudioTrim(first=afirst,last=alast,length=alength)
            break;
        except:
            adjust=adjust+1
    return  afirst, alast, adjust

def to_frame(clip, sample):
    return int((clip.video.fps/clip.audio.sample_rate)*sample)    
```

**Comment by sosie-js** [comment]
Thanks @myrsloik . 
![image](https://github.com/user-attachments/assets/a3573691-12e6-4628-be80-f0a33d097e45)

**Comment by sosie-js** [comment]
I finally released [python-bestsource 1.1](https://github.com/sosie-js/python-bestsource) fixing this problem. Big thanks for your remark @myrsloik  !
